From ece3d837bb1966c7082156c511fd2028239250d9 Mon Sep 17 00:00:00 2001
From: Frank Hunleth <fhunleth@troodon-software.com>
Date: Tue, 29 Oct 2013 14:47:10 -0400
Subject: [PATCH 2/8] erlang: multiple fixes

  1. Added staging directory to keep header files around for other
     Erlang projects to use.
  2. Scrubbed more files from the target install
---
 package/erlang/erlang.mk | 35 +++++++++++++++++++++++++++++++----
 1 file changed, 31 insertions(+), 4 deletions(-)

diff --git a/package/erlang/erlang.mk b/package/erlang/erlang.mk
index 18ceb7b..e36f03c 100644
--- a/package/erlang/erlang.mk
+++ b/package/erlang/erlang.mk
@@ -54,15 +54,42 @@ ifneq ($(BR2_PACKAGE_ERLANG_MEGACO),y)
 ERLANG_REMOVE_PACKAGES += megaco
 endif
 
-define ERLANG_REMOVE_UNUSED
-	find $(TARGET_DIR)/usr/lib/erlang -type d -name src -prune -exec rm -rf {} \;
-	find $(TARGET_DIR)/usr/lib/erlang -type d -name examples -prune -exec rm -rf {} \;
+define ERLANG_REMOVE_STAGING_UNUSED
+	# Remove intermediate build products from being copied around
+	# by Erlang release tools
+	find $(STAGING_DIR)/usr/lib/erlang -type d -name obj -prune -exec rm -rf {} \;
+
+	# Don't expose packages that won't be available on the target
+	for package in $(ERLANG_REMOVE_PACKAGES); do \
+		rm -rf $(STAGING_DIR)/usr/lib/erlang/lib/$${package}-*; \
+	done
+endef
+
+define ERLANG_REMOVE_TARGET_UNUSED
+	# Remove top level installation and header files
+	rm -rf $(TARGET_DIR)/usr/lib/erlang/misc
+	rm -f $(TARGET_DIR)/usr/lib/erlang/Install
+	rm -rf $(TARGET_DIR)/usr/lib/erlang/usr/include
+
+	# Remove intermediate build products
+	find $(TARGET_DIR)/usr/lib/erlang -type d -name obj -prune -exec rm -rf {} \;
+
+	# Remove package and erts files that are not needed
+	# at runtime
+	for dir in $(TARGET_DIR)/usr/lib/erlang/erts-* \
+		   $(TARGET_DIR)/usr/lib/erlang/lib/*; do \
+		rm -rf $${dir}/src $${dir}/c_src $${dir}/include $${dir}/doc \
+		       $${dir}/man $${dir}/examples $${dir}/emacs; \
+	done
+
+	# Remove unwanted packages
 	for package in $(ERLANG_REMOVE_PACKAGES); do \
 		rm -rf $(TARGET_DIR)/usr/lib/erlang/lib/$${package}-*; \
 	done
 endef
 
-ERLANG_POST_INSTALL_TARGET_HOOKS += ERLANG_REMOVE_UNUSED
+ERLANG_POST_INSTALL_STAGING_HOOKS += ERLANG_REMOVE_STAGING_UNUSED
+ERLANG_POST_INSTALL_TARGET_HOOKS += ERLANG_REMOVE_TARGET_UNUSED
 
 $(eval $(autotools-package))
 $(eval $(host-autotools-package))
-- 
1.8.3.2

